lib/spack/spack/binary_distribution.py

```diff
@@ -1067,15 +1070,21 @@ def generate_package_index(cache_prefix, concurrency=32):
     db.root = None
     db_root_dir = db.database_directory
 
+    ret = None
     try:
         _read_specs_and_push_index(file_list, read_fn, cache_prefix, db, db_root_dir, concurrency)
     except Exception as err:
         msg = "Encountered problem pushing package index to {0}: {1}".format(cache_prefix, err)
         tty.warn(msg)
         tty.debug("\n" + traceback.format_exc())
+        ret = err
```

Comment:
```suggestion
        raise
```

is enough

Reply:
I didn't do that because of the rmtree in the finally.